File: Check If It Is a Straight Line.cpp
class Solution {
public:
    bool collinear(vector<int> x1,vector<int> x2,vector<int> x3)
    {
        int a=x1[0]*(x2[1]-x3[1]) + x2[0]*(x3[1]-x1[1]) + x3[0]*(x1[1]-x2[1]);
        return a==0?true:false;
    }
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        if(coordinates.size()==2)
        {
            return true;
        }
        vector<int> x1=coordinates[0],x2=coordinates[1];
        for(int i=2;i<coordinates.size();i++)
        {
            if(!collinear(x1,x2,coordinates[i]))
                return false;
        }
        return true;
        
    }
};

File: Golang Solutions/Check If It Is a Straight Line.go
func checkStraightLine(coordinates [][]int) bool {
    if len(coordinates)==2{
            return true
        }
    var a int
    n:=len(coordinates)
    for i:=2;i<n;i++ {
      
        a = coordinates[0][0]*(coordinates[1][1]-coordinates[i][1]) + coordinates[1][0]*(coordinates[i][1]-coordinates[0][1]) + coordinates[i][0]*(coordinates[0][1]-coordinates[1][1])
        if a!=0{
            return false
        }
    }
    return true   
}